Reentry-To-Recovery/r2r-frontend
# Feature/RTR-37: Admin Sections

## Item Tracking

Jira link: https://reentrytorecovery.atlassian.net/browse/RTR-37

## Description

This PR implements the view, add, edit, order, and delete for course sections (topics).

## Motivation

Admin users need to be able to build courses.

## Test Plan

- Go to the admin dashboard after logging in as an admin user
- Go to edit an existing course
- You should see the course builder
- Collapsing and expanding the course builder should work as expected
- Upon expanding, you should see all the course sections along with an add new button
- Dragging the sections should allow you to reorder them
- Clicking on the sections should toggle the collapse state of the individual sections
- Upon expansion, the section content should display (note that add lesson and add quiz have not been implemented)

### Edit section
- Clicking on the edit button should open a modal to update the title
- The submit button should be disabled if no changes have been made
- If you clear the title, the input field should be in an error state and the submit button should be disabled
- If you change the title to an existing one, it should show an error toast
- Upon successfully updating the title, the modal should close and you should see a success toast
- The list in the course builder should be updated
- Note that clicking on cancel and the x button should both close the modal

### Add section
- Click on the add button to open the add modal
- The submit button should initially be disabled
- Typing into the field should enable the button
- If you clear the field, it should update to an error state with the submit button disabled
- If you change the title to an existing one and try to submit, it should show an error state
- Upon successfully adding the section, the modal should close and you should see a success toast
- The list in the course builder should be updated
- Note that clicking on cancel and the x button should both close the modal

### Delete section
- Click on the delete button to open a confirmation modal
- Canceling should close the modal
- If you choose to delete, the modal should close and you should see a success modal
- The list in the course builder should be updated

## Further Comments

Note that this only lists the content within each course section, crud for the course section content will be done in another PR.
